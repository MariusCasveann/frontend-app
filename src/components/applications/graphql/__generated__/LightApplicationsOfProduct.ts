/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: LightApplicationsOfProduct
// ====================================================

export interface LightApplicationsOfProduct_lightApplicationsOfProduct {
    __typename: 'LightApplicationDTO';
    id: any | null;
    name: string | null;
    service: boolean;
    staticPictureId: any | null;
}

export interface LightApplicationsOfProduct {
    lightApplicationsOfProduct: (LightApplicationsOfProduct_lightApplicationsOfProduct | null)[] | null;
}

export interface LightApplicationsOfProductVariables {
    productId?: any | null;
}
