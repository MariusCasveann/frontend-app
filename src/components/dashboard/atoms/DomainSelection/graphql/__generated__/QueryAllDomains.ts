/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: QueryAllDomains
// ====================================================

export interface QueryAllDomains_allDomains {
    __typename: 'Domain';
    id: any | null;
    name: string | null;
}

export interface QueryAllDomains {
    allDomains: (QueryAllDomains_allDomains | null)[] | null;
}
