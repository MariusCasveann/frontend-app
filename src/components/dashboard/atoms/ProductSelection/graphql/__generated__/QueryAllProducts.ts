/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: QueryAllProducts
// ====================================================

export interface QueryAllProducts_allProducts {
    __typename: 'LightProductDTO';
    id: any | null;
    logo: string | null;
    name: string | null;
}

export interface QueryAllProducts {
    allProducts: (QueryAllProducts_allProducts | null)[] | null;
}
