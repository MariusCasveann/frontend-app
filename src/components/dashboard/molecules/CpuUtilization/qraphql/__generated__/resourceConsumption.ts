/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import {
    DeploymentStageEnum,
    ResourceTypeEnum,
    ResourceAggregationTypeEnum,
    QueryEntity
} from './../../../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL query operation: ResourceConsumption
// ====================================================

export interface ResourceConsumption_resourceConsumption {
    __typename: 'ResourceConsumptionDTO';
    projectId: string | null;
    resourceType: string | null;
    usage: string | null;
    usageDate: any | null;
}

export interface ResourceConsumption {
    resourceConsumption: (ResourceConsumption_resourceConsumption | null)[] | null;
}

export interface ResourceConsumptionVariables {
    entityId?: any | null;
    deploymentStage?: DeploymentStageEnum | null;
    resourceType?: ResourceTypeEnum | null;
    aggregationType?: ResourceAggregationTypeEnum | null;
    entityType?: QueryEntity | null;
}
