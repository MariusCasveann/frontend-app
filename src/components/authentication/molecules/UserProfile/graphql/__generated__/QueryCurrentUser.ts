/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { PermissionTypeEnum, SecuredEntityTypeEnum } from './../../../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL query operation: QueryCurrentUser
// ====================================================

export interface QueryCurrentUser_currentUser_roles_concretePermissions {
    __typename: 'EntityPermission';
    permissionType: PermissionTypeEnum | null;
    id: any | null;
    entityType: SecuredEntityTypeEnum | null;
    entityId: any | null;
}

export interface QueryCurrentUser_currentUser_roles {
    __typename: 'Role';
    id: any | null;
    name: string | null;
    readOnly: boolean | null;
    concretePermissions: (QueryCurrentUser_currentUser_roles_concretePermissions | null)[] | null;
}

export interface QueryCurrentUser_currentUser {
    __typename: 'UserDTO';
    activated: boolean;
    createdBy: string | null;
    createdDate: any | null;
    email: string | null;
    firstName: string | null;
    id: any | null;
    imageContent: string | null;
    langKey: string | null;
    lastModifiedBy: string | null;
    lastModifiedDate: any | null;
    lastName: string | null;
    login: string | null;
    roles: (QueryCurrentUser_currentUser_roles | null)[] | null;
    staticPictureId: any | null;
    subscribedToNewsletter: boolean | null;
}

export interface QueryCurrentUser {
    currentUser: QueryCurrentUser_currentUser | null;
}
