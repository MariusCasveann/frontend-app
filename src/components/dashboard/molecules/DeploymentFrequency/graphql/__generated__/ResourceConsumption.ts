/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { ResourceTypeEnum, QueryEntity } from './../../../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL query operation: ResourceConsumption
// ====================================================

export interface ResourceConsumption_resourceConsumption {
    __typename: 'ResourceConsumptionDTO';
    projectId: string | null;
    usage: string | null;
}

export interface ResourceConsumption {
    resourceConsumption: (ResourceConsumption_resourceConsumption | null)[] | null;
}

export interface ResourceConsumptionVariables {
    entityId?: any | null;
    resourceType?: ResourceTypeEnum | null;
    entityType?: QueryEntity | null;
    toDate?: any | null;
    fromDate?: any | null;
}
