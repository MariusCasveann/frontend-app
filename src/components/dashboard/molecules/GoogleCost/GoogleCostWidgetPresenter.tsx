import React from 'react';
import * as chartjs from 'chart.js';
import { Bar } from 'react-chartjs-2';
import LoadingOrErrorPresenter from '../../../common/organisms/LoadingOrError/LoadingOrErrorPresenter';
import { hasChartData } from '../../helpers/checkChartData';
import NumberDisplay, { NumberType } from '../../../common/molecules/NumberDisplay/NumberDisplay';
import { DeploymentStageEnum } from '../../../../model/__generated__/globalTypes';
import './GoogleCostPresenter.css';
import { MESSAGE_WHEN_NO_DATA } from '../../../../utils/constants';

interface GoogleCostWidgetPresenterProps {
    loading: boolean;
    error: string;
    chartData?: chartjs.ChartData;
    sumMonth: number;
    maxValue: number;
    stage: DeploymentStageEnum;
}

export default ({ chartData, error, loading, sumMonth, stage, maxValue }: GoogleCostWidgetPresenterProps) => {
    if (loading || error) {
        return <LoadingOrErrorPresenter error={error} loading={loading} />;
    }
    if (hasChartData(chartData)) {
        return (
            <>
                <div className="chart">
                    {chartData && (
                        <Bar
                            data={chartData}
                            options={{
                                responsive: true,
                                maintainAspectRatio: false,
                                legend: {
                                    display: false
                                },
                                scales: {
                                    yAxes: [
                                        {
                                            ticks: {
                                                max: maxValue
                                            }
                                        }
                                    ]
                                }
                            }}
                        />
                    )}
                </div>
                <div className="numbers-container">
                    <NumberDisplay value={sumMonth} type={NumberType.currency} label={stage} />
                </div>
            </>
        );
    } else if (hasChartData(chartData)) {
        return <p>{MESSAGE_WHEN_NO_DATA}</p>;
    }

    return null;
};
