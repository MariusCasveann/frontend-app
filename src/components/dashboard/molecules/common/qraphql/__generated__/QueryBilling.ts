/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { QueryEntity } from './../../../../../../model/__generated__/globalTypes';

// ====================================================
// GraphQL query operation: QueryBilling
// ====================================================

export interface QueryBilling_billingQuery {
    __typename: 'GoogleBillingResultDTO';
    resourceType: string | null;
    cost: any | null;
    date: any | null;
    currency: string | null;
}

export interface QueryBilling {
    billingQuery: (QueryBilling_billingQuery | null)[] | null;
}

export interface QueryBillingVariables {
    entity?: QueryEntity | null;
    entityId?: number | null;
    stage?: string | null;
    financialYear?: number | null;
    toDate?: any | null;
    fromDate?: any | null;
}
