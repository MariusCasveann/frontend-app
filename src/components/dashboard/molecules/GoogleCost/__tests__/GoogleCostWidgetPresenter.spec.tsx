import * as React from 'react';
import renderer from 'react-test-renderer';
jest.mock('react-chartjs-2', () => ({ Bar: 'Bar' }));
jest.mock('../../../../common/organisms/LoadingOrError/LoadingOrErrorPresenter', () => 'LoadingOrErrorPresenter');
import GoogleCostWidgetPresenter from '../GoogleCostWidgetPresenter';
import { DeploymentStageEnum } from '../../../../../model/__generated__/globalTypes';

const mockData = {
    labels: ['Oct.', 'Nov.', 'Dec.', 'Jan.', 'Feb.', 'Mar.', 'Apr.', 'May', 'Jun.', 'Jul.', 'Aug.', 'Sep.'],
    datasets: [
        {
            data: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12],
            backgroundColor: ['#859ec4']
        }
    ]
};

const mockMaxValue = 80.0;

describe('GoogleCostWidgetPresenter', () => {
    describe('when there are chart data', () => {
        it('renders a Bar chart', () => {
            const wrapper = renderer
                .create(
                    <GoogleCostWidgetPresenter
                        maxValue={mockMaxValue}
                        error=""
                        loading={false}
                        chartData={mockData}
                        sumMonth={78}
                        stage={DeploymentStageEnum.DEV}
                    />
                )
                .toJSON();
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when chart data is undefined', () => {
        it('renders an error', () => {
            const wrapper = renderer
                .create(
                    <GoogleCostWidgetPresenter
                        maxValue={mockMaxValue}
                        error="Error"
                        loading={false}
                        chartData={undefined}
                        sumMonth={78}
                        stage={DeploymentStageEnum.DEV}
                    />
                )
                .toJSON();
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when chart data is undefined', () => {
        it('renders loading', () => {
            const wrapper = renderer
                .create(
                    <GoogleCostWidgetPresenter
                        maxValue={mockMaxValue}
                        error=""
                        loading={true}
                        chartData={undefined}
                        sumMonth={78}
                        stage={DeploymentStageEnum.DEV}
                    />
                )
                .toJSON();
            expect(wrapper).toMatchSnapshot();
        });
    });
});
