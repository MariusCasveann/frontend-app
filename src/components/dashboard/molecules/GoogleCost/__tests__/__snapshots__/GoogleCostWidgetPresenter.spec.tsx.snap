// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleCostWidgetPresenter when chart data is undefined renders an error 1`] = `
<LoadingOrErrorPresenter
  error="Error"
  loading={false}
/>
`;

exports[`GoogleCostWidgetPresenter when chart data is undefined renders loading 1`] = `
<LoadingOrErrorPresenter
  error=""
  loading={true}
/>
`;

exports[`GoogleCostWidgetPresenter when there are chart data renders a Bar chart 1`] = `
Array [
  <div
    className="chart"
  >
    <Bar
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": Array [
                "#859ec4",
              ],
              "data": Array [
                1,
                2,
                3,
                4,
                5,
                6,
                7,
                8,
                9,
                10,
                12,
              ],
            },
          ],
          "labels": Array [
            "Oct.",
            "Nov.",
            "Dec.",
            "Jan.",
            "Feb.",
            "Mar.",
            "Apr.",
            "May",
            "Jun.",
            "Jul.",
            "Aug.",
            "Sep.",
          ],
        }
      }
      options={
        Object {
          "legend": Object {
            "display": false,
          },
          "maintainAspectRatio": false,
          "responsive": true,
          "scales": Object {
            "yAxes": Array [
              Object {
                "ticks": Object {
                  "max": 80,
                },
              },
            ],
          },
        }
      }
    />
  </div>,
  <div
    className="numbers-container"
  >
    <div
      className="number-display"
    >
      <h2
        aria-label={null}
        className="ant-typography number-display-number"
        style={
          Object {
            "WebkitLineClamp": null,
          }
        }
      >
        78
         EUR
         
      </h2>
      <span
        aria-label={null}
        className="ant-typography number-display-label"
        style={
          Object {
            "WebkitLineClamp": null,
          }
        }
      >
        DEV
      </span>
    </div>
  </div>,
]
`;
